.github/workflows/build-and-test.yml

```
@@ -0,0 +1,305 @@
+    - name: Build Tester
+      run: make -C tests BUILD_PROFILE=release driver
+    - uses: actions/upload-artifact@master
+      with:
+        name: hermetic-driver
+        path: ./tests/target/release/hermetic-driver
+        retention-days: 1
+
+  run-integration-tests:
+    name: Run Integration Tests
+    runs-on: ubuntu-latest
+    environment: dev-qa-2024
+    needs:
+      - generate-matrix
+      - build-driver
+      - publish-suite
+      - build
+    strategy:
+      fail-fast: false
+      matrix:
+        networks: ${{ fromJSON(needs.generate-matrix.outputs.networks) }}
+    steps:
+      -
+        name: Checkout
+        uses: actions/checkout@v3
+
+      -
+        name: Setup GKE auth
+        uses: 'google-github-actions/auth@v1'
+        with:
+          credentials_json: ${{ secrets.GKE_SA_KEY }}
+      -
+        name: Get GKE credentials
+        uses: 'google-github-actions/get-gke-credentials@v1'
+        with:
+          cluster_name: ${{ vars.GKE_CLUSTER }}
+          location: ${{ vars.GKE_ZONE }}
+      - uses: actions/download-artifact@master
+        with:
+          name: hermetic-driver
+          path: ./bin
+      -
+        name: Test ${{ matrix.networks }}
+        run: |
+          set -euxo pipefail
+          export BUILD_TAG=${{ needs.generate-matrix.outputs.build_tag }}
+          export TEST_NETWORK=./tests/networks/${{ matrix.networks }}.yaml
+          chmod +x ./bin/hermetic-driver
+
+          # Setup network yaml to be use the ceramic-one image build for this run (i.e. its git sha)
+          RUST_CERAMIC_IMAGE_TAG='${{ needs.build.outputs.image_tag }}'
+          RUST_CERAMIC_IMAGE_REPO='public.ecr.aws/r5b3e0r5/3box/ceramic-one'
+          export RUST_CERAMIC_IMAGE=${RUST_CERAMIC_IMAGE_REPO}:${RUST_CERAMIC_IMAGE_TAG}
+          export KERAMIK_NETWORK_NAME=hermetic-rust-ceramic-${BUILD_TAG}
+          curl -L https://github.com/mikefarah/yq/releases/download/v4.40.5/yq_linux_amd64 -o ./bin/yq
+          chmod +x ./bin/yq
+          ./bin/yq -i '.spec.ceramic[0].ipfs.rust.image = strenv(RUST_CERAMIC_IMAGE)' $TEST_NETWORK
+          ./bin/yq -i '.metadata.name = strenv(KERAMIK_NETWORK_NAME)' $TEST_NETWORK
+          ./bin/yq --no-colors $TEST_NETWORK
+
+          # For scheduled events, "test_selector" will be null, so default to running all tests.
+          test_selector=${{ github.event.inputs.test_selector || 'correctness' }}
+
+          # For PR and merge group events, run only "fast" tests.
+          # Note it is not possible to override "test_selector" for these types of events.
+          if [[ ${{ github.event_name == 'pull_request' || github.event_name == 'merge_group' }} == 'true' ]]; then
+              test_selector="correctness/fast"
+          fi
+          make -C tests TEST_SELECTOR="$test_selector" HERMETIC_CMD=../bin/hermetic-driver hermetic-tests
+  run-migration-tests:
+    name: Run Migration Tests
+    runs-on: ubuntu-latest
+    environment: dev-qa-2024
```

Can move this to tnet/prod cluster now.
```suggestion
    environment: tnet-prod-2024
```